tsl/src/fdw/deparse.c

```diff
@@ -834,6 +834,169 @@ deparseSelectStmtForRel(StringInfo buf, PlannerInfo *root, RelOptInfo *rel, List
 	deparseLockingClause(&context);
 }
 
+/*
+ * Construct "SELECT DISTINCT target_list" or "SELECT DISTINCT ON (col1, col..)
+ * target_list" statement to push down the DISTINCT clause to the remote side.
+ *
+ * We only allow references to basic "Vars" or constants in the DISTINCT exprs
+ *
+ * So, "SELECT DISTINCT col1" is fine but "SELECT DISTINCT 2*col1" is not.
+ *
+ * "SELECT DISTINCT col1, 'const1', NULL, col2" which is a mix of column
+ * references and constants is also supported. Everything else is not supported.
+ *
+ * Note that "SELECT DISTINCT col1, col2" is different from "SELECT DISTINCT
+ * col2, col1". So, the attributes have to be fetched in the same order as
+ * well from the remote side. This means that "retrieved_attrs" list also
+ * needs to be constructed in the same order. This function returns a
+ * "dc_varattnos" list which points to the attributes as per the DISTINCT
+ * specification.
+ */
+static void
+deparseDistinctClause(StringInfo buf, List **dc_varattnos, deparse_expr_cxt *context)
```

Comment:
We don't use CamelCase for function names in our code style, so a conforming name would be:

```suggestion
deparse_distinct_clause(StringInfo buf, List **dc_varattnos, deparse_expr_cxt *context)
```
I realize there are other functions in this file that is not conforming because they were copied more-or-less verbatim from the PostgreSQL FDW code.

Reply:
i think we should stick to postgres style for this file, the distinct deparsing might actually be a patch we could supply upstream

Reply:
Agreed with Sven here.